### PR TITLE
contributing: Add contributing to documentation guidelines

### PR DESCRIPTION
**Pull request by StefanJum**
Add a page for contributing to documentation, building the website, using the superlinter, etc.
Most of this page is copied from the old docs.

### CODE REVIEW

File: content/docs/contributing/docs.mdx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+npm install textlint-rule-terminology
+npm install textlint-rule-no-todo
+```
+
+Use the linters with `npx` from the documentation directory:
+
+```console
+npx markdownlint --config .github/workflows/config/config.json --rules .github/workflows/rules/rules.js path/to/markdown/file.mdx
+npx textlint --config .github/workflows/.textlintrc --rule <rule-name> path/to/markdown/file.mdx
+npx jscpd <path> # Tipically a directory
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:215]:
```suggestion
npx jscpd <path> # Typically a directory
```

**Review by razvand — COMMENTED** [review]
@StefanJum , see small comment.

File: content/docs/contributing/docs.mdx [hunk 2]
```diff
@@ -0,0 +1,274 @@
+---
+title: Contributing to Documentation
+---
+
+## Documentation
+
+As with the other parts of Unikraft, we welcome contributions to the [documentation repository](https://github.com/unikraft/docs).
+Documentation is written using [GitHub Markdown](https://github.github.com/gfm/).
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:8]: Isn't it now [MDX](https://mdxjs.com/)?

File: content/docs/contributing/docs.mdx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+We follow the [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow) for contributions.
+Submit corrections and improvements to documentation as pull requests (PRs) that we will then review and approve in the documentation.
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:11]:
```suggestion
Submit new content, corrections and improvements to documentation as pull requests (PRs) that we will then review and approve in the documentation.
```

File: content/docs/contributing/docs.mdx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+You can add entire sections if you think a topic is missing or update existing ones with missing information.
+
+You can also open up an issue to signal a problem or a missing part in the documentation that you request someone to fix.
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:14]:
```suggestion
You can also open up [an issue](https://github.com/unikraft/docs/issues) to signal a problem or a missing part in the documentation that you request someone to fix.
```

File: content/docs/contributing/docs.mdx [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+content
+|-- blog/                   # Blog posts, <date>-<blogpost title>.md
+|-- community/              # Community related content (events, general information, etc.)
+|-- docs/                   # Unikraft documentation, split by categories into subdirectories
+|-- guides/                 # Short and on point guides, good for getting started
+|-- hackathons/             # Hackathon landing pages
+|-- imprint.md
+|-- privacy.md
+`-- releases/               # Unikraft releases summaries
+```
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:39]: This needs to be updated to the current layout.

File: content/docs/contributing/docs.mdx [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+```console
+docker build -t ghcr.io/unikraft/docs:base --target base .
+
+docker run -it --rm -v $(pwd):/docs -w /docs -p 3000:3000 --entrypoint sh ghcr.io/unikraft/docs:base
+
+yarn run dev
+```
+
+You may need admin privileges to run the first 2 commands.
+The website will be rendered at `localhost:3000`.
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:79]:
```suggestion
Open up your browser and point to http://localhost:3000 to check the deployed documentation website.

If you do updates the documentation files (even outside the Docker environment) they will be picked up and rendered at http://localhost:3000.
```

File: content/docs/contributing/docs.mdx [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+</Info>
+
+Then, to build and deploy the site, run:
+
+```console
+yarn install
+yarn run dev
+```
+
+Access the `http://localhost:3000/` URL in the browser to view the site.
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:99]:
```suggestion
As above, open up your browser and point to http://localhost:3000 to check the deployed documentation website.
```

File: content/docs/contributing/docs.mdx [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+
+#### Using the Linter
+
+To automate a big part of the reviewing process, we use the [`super-linter`](https://github.com/github/super-linter) on the `docs` repository.
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:103]:
```suggestion
To automate a big part of the reviewing process, we use the [`super-linter`](https://github.com/github/super-linter) on this repository.
```

**Review by razvand — CHANGES_REQUESTED** [review]
@StefanJum , some extra items.

File: content/docs/contributing/docs.mdx (hunk 5 above)
Comment (StefanJum) [reply content/docs/contributing/docs.mdx:39]: This is the current layout

File: content/docs/contributing/docs.mdx (hunk 8 above)
Comment (StefanJum) [reply content/docs/contributing/docs.mdx:103]: The content will be viewed on the website, `this` would seem like we are talking about the website, not a repository. I would leave it as it is now, if you want me to change it let me know.

File: content/docs/contributing/docs.mdx [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+Submit new content, corrections and improvements to documentation as pull requests (PRs) that we will then review and approve in the documentation.
+You can add entire sections if you think a topic is missing or update existing ones with missing information.
+
+You can also open up [an issue](https://github.com/unikraft/docs/issues) to signal a problem or a missing part in the documentation that you request someone to fix.
+
+We use English as the documentation language.
+Please proofread your text before submitting a pull request.
+
+Write each sentence on a new line, since it make the content easy to review and modify.
+This way, changing one sentence only affects one line in the source code.
```
Comment (razvand) [inline content/docs/contributing/docs.mdx:20]:
```suggestion
This way, changing one sentence only affects one line in the source code, and it is easier to follow the content and to reviews.
```

**Review by razvand — APPROVED** [review]
@StefanJum , just a small change and then we can merge it.

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**